src/analyses/tmpSpecial.ml

```diff
@@ -82,9 +120,22 @@ struct
 
   let query ctx (type a) (q: a Queries.t) : a Queries.result =
     match q with
-    | TmpSpecial lv -> let ml = fst (D.find lv ctx.local) in
+    | TmpSpecial lv -> let ml = fst (MM.find lv (fst ctx.local)) in
       if ML.is_bot ml then Queries.Result.top q
       else ml
+    | TmpSpecialInv lv -> (
+        if M.tracing then M.trace "tmpSpecialInv" "MInv.find lv: %s\n" (fst lv).vname;
+        match MInv.find lv (snd ctx.local) with
+        | `Top -> Queries.Result.top q
+        | es ->
+          let mlv = MInvEntrySet.fold (fun (v, ml, deps) acc -> Queries.MLInv.add (v,ml) acc) es (Queries.MLInv.empty ())
+                    |> Queries.MLInv.filter (fun (v,ml) ->
+                        match ml, v with
+                        | `Bot, _ -> false
+                        | _, `Lifted exp -> true
+                        | _ -> false) in
```

Comment:
This way of doing looks as if it may be in a figure of speech potentially trying to achieve the goal in a way that might be described as roundabout. You might want to do the filtering right in place. :wink: